compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -1847,19 +1847,13 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                     path.segments[..path.segments.len() - 2].iter(),
                     GenericsArgsErrExtend::None,
                 );
-                // HACK: until we support `<Type as ~const Trait>`, assume all of them are.
```

Comment:
Why was this removed? Do we not need this anymore?

Reply:
since we don't have the constness param on traits, we don't need <T as ~const> anymore